modules/core/src/com/haulmont/cuba/core/app/scheduling/Scheduling.java

```diff
@@ -421,4 +421,19 @@ protected boolean isRunning(ScheduledTask task) {
         }
         return false;
     }
+
+    @Override
+    public void runOnce(ScheduledTask task) throws RuntimeException {
+        try {
+            Integer priority = getServerPriority(task, serverInfo.getServerId());
+            if (priority == null) {
+                throw new RuntimeException(task + ": not in permitted hosts or not a master");
+            }
+
+            lastStartCache.put(task, timeSource.currentTimeMillis());
+            runner.runTaskOnce(task, timeSource.currentTimeMillis(), getUserSession(task));
+        } catch (LoginException e) {
```

Comment:
Why do you catch LoginException?

Reply:
Because I get the user session and use it in SecurityContext when run a task